edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/PodExtensions.cs

```diff
@@ -0,0 +1,93 @@
+using k8s.Models;
+using Microsoft.Azure.Devices.Edge.Agent.Core;
+using Microsoft.Azure.Devices.Edge.Util;
+using System;
+using AgentDocker = Microsoft.Azure.Devices.Edge.Agent.Docker;
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes
+{
+    public static class PodExtensions
```

Comment:
>PodExtensions [](start = 24, length = 13)

Some of these routines needed improvement from preview -> master, mostly due to me actually testing the code. :) Please confirm behavior is the same -- I'll check it when this PR gets out of draft.